packages/components/organisms/f-checkout/src/components/AgeVerification.vue

```diff
@@ -0,0 +1,164 @@
+                is-full-width
+                data-test-id="age-verification-redirect-button"
+                @click.native="handleAgeVerifcation">
+                {{ $t(`ageVerification.buttonText`) }}
+            </f-button>
+        </form>
+    </card>
+</template>
+
+<script>
+import Card from '@justeat/f-card';
+import '@justeat/f-card/dist/f-card.css';
+import FButton from '@justeat/f-button';
+import '@justeat/f-button/dist/f-button.css';
+import FormField from '@justeat/f-form-field';
+import '@justeat/f-form-field/dist/f-form-field.css';
+import { mapState, mapActions } from 'vuex';
+import { VUEX_CHECKOUT_MODULE } from '../constants';
+
+export default {
+    components: {
+        Card,
+        FButton,
+        FormField
+    },
+
+    data () {
+        return {
+            selectedDate: {
+                day: '',
+                month: '',
+                year: ''
+            }
+        };
+    },
+
+    computed: {
+        ...mapState(VUEX_CHECKOUT_MODULE, ['customer']),
+
+        days () {
+            return Array.from({ length: 31 }, (_, index) => {
+                const day = (index + 1).toString();
+
+                return {
+                    text: day,
+                    value: day
+                };
+            });
+        },
+
+        months () {
+            return new Array(12).fill(0).map((_, i) => {
+                const date = new Date(`${i + 1}/1`);
+
+                return {
+                    text: date.toLocaleDateString(undefined, { month: 'long' }),
+                    value: date.getMonth()
+                };
+            });
+        },
+
+        years () {
+            const currentYear = new Date().getFullYear();
+            const start = currentYear - 10;
+            const end = currentYear - 100;
+
+            return Array(start - end)
+                    .fill('')
+                    .map((_, index) => {
+                        const year = (start - index).toString();
+
+                        return {
+                            text: year,
+                            value: year
+                        };
+                    });
+        }
+    },
+
+    mounted () {
+        this.initDateOfBirth();
+    },
+
+    methods: {
+        ...mapActions(VUEX_CHECKOUT_MODULE, ['updateDateOfBirth']),
+
+        initDateOfBirth () {
+            this.selectedDate.day = this.customer.dateOfBirth?.getDate().toString() || this.days[0].text;
+            this.selectedDate.month = this.customer.dateOfBirth?.getMonth().toString() || this.months[0].value.toString();
+            this.selectedDate.year = this.customer.dateOfBirth?.getFullYear().toString() || this.years[0].text;
+        },
+
+        selectionChanged (selection, type) {
+            this.selectedDate[type] = selection;
+        },
+
+        handleAgeVerifcation () {
+            const dateOfBirth = new Date(this.selectedDate.year, this.selectedDate.month, this.selectedDate.day);
```

Comment:
if the above is true this will output the next available that has a 31 day month if its a 30 day month.

Reply:
based On outside communication I understand this will be resolved in a later phase and won't be touching production in this state.